### DISCUSSION TITLE
[idea] Add a window for changing ChatGPT parameters

### DISCUSSION THREAD

**Discussion by FBorowiec** [jackMort/ChatGPT.nvim#17] [Ideas]
Hi! First let me thank you for your excellent work on this plugin! :)

Would it be possible to add a window for easier change of the OpenAI parameters?
```lua
        openai_params = {
            model = "text-davinci-003",
            frequency_penalty = 0,
            presence_penalty = 0,
            max_tokens = 300,
            temperature = 1,
            top_p = 1,
            n = 1,
        },
```
Thanks!

**Comment by roccho-dev** [comment]
Too I want.

Or if I want to change it manually, is this place where I should modify?
![image](https://user-images.githubusercontent.com/40359643/211276781-7c512220-956d-4bff-aadd-dd4b0f85d29e.png)

**Reply by jackMort** [reply]
For quick action there are predefined parameters without option to modify it but for chat window you can  change it in config as in README or open config window using <C-o> shortcut

**Reply by roccho-dev** [reply]
> For quick action there are predefined parameters without option to modify it but for chat window you can change it in config as in README or open config window using shortcut

Thanks.

**Comment by 00sapo** [comment]
I think this would be helpful. Generally, a temperature of 0.2-0.3 is ok, but sometimes, when the model returns wrong results, a higher temperature is needed. It should be modifiable from the edit code window... The same also applies to the model: sometimes we want to use the  gpt-3.5-turbo model in chat mode from the "edit code" window...

**Comment by jackMort** [comment]
`<C-o>` is opening the sessions and settings panels

**Reply by 00sapo** [reply]
I know, but is it possible to change the settings at runtime?

**Reply by 00sapo** [reply]
Just found it by myself: pressing enter in the setting window allows for editing that parameter... I will try to document this and actions.json file
